### PR TITLE
Adds glowcaps

### PR DESCRIPTION
**Pull request by Incoming5643**
d10_icon

Glowcaps are red mutant glowshrooms. They can be used in all the same ways normal glowshrooms can, but they also boast a special feature.

When you finish eating a glowcap, the batteries of any electronics you have on/in you are recharged up to potency%. A great gift to give to the warden, or anyone else who might be packing laser weaponry.

Keep in mind the effect is only given after you FINISH the glowcap, so at high potencies you'll either have to drain the radium out ahead of time or resign yourself to taking a few bites before you actually finish your meal.

:cl: incoming5643
rscadd: a mutation of glowshrooms, glowcaps, have recently been discovered in botany.
/:cl:

### CODE REVIEW

**Comment by Iamgoofball** [comment]
tumblr_npnaqnb2jb1tg07vxo1_1280

**Comment by Incoming5643** [comment]
<a href="https://www.youtube.com/watch?v=COdkHuogVkw">SHHHHHHHHH</a>

**Comment by newfren** [comment]
:+1:

**Comment by calibraptor** [comment]
yes

File: code/modules/hydroponics/grown.dm [hunk 1]
```diff
@@ -1218,6 +1218,27 @@ obj/item/weapon/reagent_containers/food/snacks/grown/shell/eggy/add_juice()
...
+	name = "glowcap cluster"
+	desc = "<I>Mycena Ruthenia</I>: This species of mushroom glows in the dark, but aren't bioluminescent. They're warm to the touch..."
+	icon_state = "glowcap"
+	filling_color = "#00FA9A"
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/mushroom/glowshroom/glowcap/On_Consume()
+	if(!reagents.total_volume)
+		var/batteries_recharged = 0
+		for(var/obj/item/weapon/stock_parts/cell/C in usr.GetAllContents())
+			var/newcharge = (potency/100)*C.maxcharge
```
Comment (duncathan) [inline code/modules/hydroponics/grown.dm]: make it `(potency*0.01)`

Comment (Incoming5643) [inline code/modules/hydroponics/grown.dm]: done

**Comment by Fox-McCloud** [comment]
One thing to note about this; if your cell's charge is over how much it would restore, this makes the glowcap do absolutely nothing; is that intentional? Wouldn't it be better to add charge to everything as opposed to setting the charge to something?

**Comment by Incoming5643** [comment]
yes that's intentional. I was afraid of making them too potent in terms of just holding onto a bunch (painlessly low radium content) low potency glowcaps for a full ghetto backpack recharge. If you're gonna use the tiny shrooms, all it will do is give you one free shot from a drained battery.

**Comment by Fox-McCloud** [comment]
@Incoming5643 

I suspected as such; just wanted to make sure. Great PR by the by.

**Pull request merged**